index.js

```diff
@@ -47,13 +50,18 @@ module.exports = function (source, options) {
 	// reassign options.sourcemap boolean to one of our two acceptable Sass arguments
 	options.sourcemap = options.sourcemap === true ? 'file' : 'none';
 
-	// sass options need unix style slashes
-	tempDir = slash(path.join(TMP_DIR, options.container));
+	// Create hash which is based on source and options, so that  same source with different options
+	// won't be stored in same temporary directory
+	sourceHash = md5(source + JSON.stringify(options));
+	tempDir = path.join(projectTempDir, sourceHash);
+
+	// Store sass cache to temporary directory
+	options['cache-location'] = path.join(tempDir, '.sass-cache');
```

Comment:
Thanks Sindre. Have been work only for a month or two.

On Saturday, July 25, 2015, Sindre Sorhus <notifications@github.com> wrote:

> In index.js
> <https://github.com/sindresorhus/gulp-ruby-sass/pull/233#discussion_r35486419>
> :
>
> > @@ -47,13 +50,18 @@ module.exports = function (source, options) {
> >  	// reassign options.sourcemap boolean to one of our two acceptable Sass arguments
> >  	options.sourcemap = options.sourcemap === true ? 'file' : 'none';
> >
> > -	// sass options need unix style slashes
> > -	tempDir = slash(path.join(TMP_DIR, options.container));
> > +	// Create hash which is based on source and options, so that  same source with different options
> > +	// won't be stored in same temporary directory
> > +	sourceHash = md5(source + JSON.stringify(options));
> > +	tempDir = path.join(projectTempDir, sourceHash);
> > +
> > +	// Store sass cache to temporary directory
> > +	options['cache-location'] = path.join(tempDir, '.sass-cache');
>
> Option name should be camelCase.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/sindresorhus/gulp-ruby-sass/pull/233/files#r35486419>.
>


-- 
Rob Wierzbowski
@robwierzbowski <http://twitter.com/#!/robwierzbowski>
http://github.com/robwierzbowski
http://robwierzbowski.com